alpha/apps/kaltura/lib/events/consumers/kObjectReadyForIndexInheritedTreeEventConsumer.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * Applicative event that raised by the developer when object is ready for update
+ */
+interface kObjectReadyForIndexInheritedTreeEventConsumer extends KalturaEventConsumer
+{
+	/**
+	 * @param BaseObject $object
+	 * @param array $partnerCriteriaParams
+	 * @param BatchJob $raisedJob
+	 * @return bool true if should continue to the next consumer
+	 */
+	public function objectReadyForIndexInheritedTreeEvent(BaseObject $object, array $partnerCriteriaParams, BatchJob $raisedJob = null);
```

Comment:
Since the $partnerCriteriaParams is relevant to any multi-deferred event, I think it should be implemented as an interface with setPartnerCriteriaParams method.
If you follow this path, you won't have to add isMulti argument to the raiseDeferredEvent method because you'll be able to check if the event is instance of the multi-deferred interface.